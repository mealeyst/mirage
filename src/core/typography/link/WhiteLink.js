import PropTypes from 'prop-types'
import styled from 'styled-components'

import BaseLink from './Link.base'

const WhiteLink = styled(BaseLink)`
  color:  ${props => props.theme.colors.white};
  &:active, &:focus, &:hover, &:visited {
    color: ${props => props.theme.colors.white};
    ${props => props.underline ?
      `border-bottom-color: ${props.theme.colors.white};` :
      ''
    }
  }
`

WhiteLink.propTypes = {
  theme: PropTypes.shape({
    colors: PropTypes.shape({
      white: PropTypes.string
    })
  }),
  underline: PropTypes.bool.isRequired
}

WhiteLink.defaultProps = {
  underline: true
}

/** @component */
export default WhiteLink
