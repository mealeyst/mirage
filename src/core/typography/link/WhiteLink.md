```js
<div style={{backgroundColor: '#333', padding: '15px'}}>
  <WhiteLink href="#">Example White Link</WhiteLink>
</div>
```

# No Hover Underline Example:
```js
<div style={{backgroundColor: '#333', padding: '15px'}}>
  <WhiteLink underline={false} href="#">Example White Link</WhiteLink>
</div>
```

# Upper Case Case Example:
```js
<div style={{backgroundColor: '#333', padding: '15px'}}>
  <WhiteLink uppercase href="#">Example White Link</WhiteLink>
</div>
```
