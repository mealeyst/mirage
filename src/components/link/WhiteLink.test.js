import React from 'react'
import { css } from 'styled-components'
import 'jest-styled-components'

import WhiteLink from './WhiteLink'

const { mountWithTheme } = global

describe('(Styled Component) WhiteLink', () => {
  const createWhiteLink = (props) => {
    return mountWithTheme(<WhiteLink {...props} />)
  }

  test('matching the snapshot', () => {
    expect(createWhiteLink())
    .toMatchSnapshot()
  })

  test('not passing the underline prop', () => {
    expect(createWhiteLink({underline: false}))
    .toHaveStyleRule({'border-bottom-color': undefined})
  })
})
