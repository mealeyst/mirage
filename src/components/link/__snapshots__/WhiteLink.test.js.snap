// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Styled Component) WhiteLink matching the snapshot 1`] = `
.c1 {
  -webkit-transition: color 0.25s ease-in-out,border-bottom-color 0.25s ease-in-out;
  transition: color 0.25s ease-in-out,border-bottom-color 0.25s ease-in-out;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: initial;
  border-bottom: 0.2rem solid;
  border-bottom-color: transparent;
  font-family: "ff-din-web",Helvetica,Arial,sans-serif;
  font-size: 1.4rem;
  font-weight: 500;
}

.c1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  pointer: cursor;
}

.c0 {
  color: #FFFFFF;
}

.c0:active,
.c0:focus,
.c0:hover,
.c0:visited {
  color: #FFFFFF;
  border-bottom-color: #FFFFFF;
}

<Styled(styled.a)
  underline={true}
>
  <styled.a
    className="c0"
    underline={true}
    uppercase={false}
  >
    <a
      className="c0 c1"
    />
  </styled.a>
</Styled(styled.a)>
`;
